giraph-core/src/main/java/org/apache/giraph/partition/ByteArrayPartition.java

```diff
@@ -233,14 +232,8 @@ public void write(DataOutput output) throws IOException {
       entry.getKey().write(output);
       // Note here that we are writing the size of the vertex data first
       // as it is encoded in the first four bytes of the byte[]
-      int vertexDataSize;
-      if (useUnsafeSerialization) {
```

Comment:
Are we not supporting the non-unsafe version anymore?

Reply:
I have added back the non-unsafe version